pkg/platform/pullrequest.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package platform
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/abcxyz/guardian/pkg/github"
+)
+
+var _ ChangeRequest = (*PullRequest)(nil)
+
+// PullRequestInput defines the required inputs for creating a new PullRequest
+// instance.
+type PullRequestInput struct {
+	GitHubToken string
+
+	GitHubAppID             string
+	GitHubAppPrivateKeyPEM  string
+	GitHubAppInstallationID string
+
+	Owner             string
+	Repository        string
+	PullRequestNumber int
+}
+
+// PullRequest is the GitHub implementation for the ChangeRequest interface.
+type PullRequest struct {
+	client github.GitHub
+	params *PullRequestInput
+}
+
+// NewPullRequest creates a new PullRequest instance.
+func NewPullRequest(ctx context.Context, inputs *PullRequestInput) (*PullRequest, error) {
+	tokenSource, err := github.TokenSource(ctx, &github.TokenSourceInputs{
```

Comment:
I think this is gonna conflict with some of the inflight changes that @verbanicm has. Can you two sync on an order of operations?

Reply:
Lets get this merged first and ill rebase my changes on to this. I started fixing the other commands so this is just one more.